<launch>
    <arg name="first_tb3"  default="hider"/>
    <arg name="second_tb3" default="seeker"/>
    <arg name="no_static_map" default="true"/>
    <arg name="base_global_planner" default="navfn/NavfnROS"/>
    <arg name="base_local_planner" default="dwa_local_planner/DWAPlannerROS"/>
    <node pkg="tf" type="static_transform_publisher" name="map_to_odom" args="0 0 0 0 0 0 /map /odom 10" />
    
    <node pkg="hide_and_seek" type="referee_node.py" name="referee_node" output="screen">
        <param name="game_duration" value="60" />
        <param name="caught_distance" value="1.0" />
        <param name="preparation_time" value="15" />
    </node>

    <group ns="hider">
        <node pkg="hide_and_seek" type="odom_to_baselink.py" name="odom_to_baselink" output="screen">
            <param name="robot_name" value="$(arg first_tb3)" />
        </node>
        <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" >

            <param name="base_global_planner" value="$(arg base_global_planner)"/>
            <param name="base_local_planner" value="$(arg base_local_planner)"/>  
            <!-- <rosparam file="$(find hide_and_seek)/config/planner.yaml" command="load"/> -->
        
            <rosparam file="$(find hide_and_seek)/config/costmap_common_$(arg first_tb3).yaml" command="load" ns="global_costmap" />
            <rosparam file="$(find hide_and_seek)/config/costmap_common_$(arg first_tb3).yaml" command="load" ns="local_costmap" />
        
            <rosparam file="$(find hide_and_seek)/config/costmap_local.yaml" command="load" ns="local_costmap" />
            <param name="local_costmap/width" value="20.0"/> <!-- 10.0 --> 
            <param name="local_costmap/height" value="20.0"/> <!-- 10.0 -->
        
            <!-- <rosparam file="$(find hide_and_seek)/config/costmap_global_static.yaml" command="load" ns="global_costmap" unless="$(arg no_static_map)"/> -->
            
            <rosparam file="$(find hide_and_seek)/config/costmap_global_laser.yaml" command="load" ns="global_costmap" if="$(arg no_static_map)"/>
            <param name="global_costmap/width" value="20.0" if="$(arg no_static_map)"/>
            <param name="global_costmap/height" value="20.0" if="$(arg no_static_map)"/>
        
            <remap from="/odom" to="$(arg first_tb3)/odom" />
            <remap from="/cmd_vel" to="$(arg first_tb3)/cmd_vel" />
        
          </node>
          <node pkg="hide_and_seek" type="hider_bot.py" name="hider_node" output="screen"/>
    </group>
    <group ns="seeker">
        <node pkg="hide_and_seek" type="odom_to_baselink.py" name="odom_to_baselink" output="screen">
            <param name="robot_name" value="$(arg second_tb3)" />
        </node>
        <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" >

            <param name="base_global_planner" value="$(arg base_global_planner)"/>
            <param name="base_local_planner" value="$(arg base_local_planner)"/>  
            <!-- <rosparam file="$(find hide_and_seek)/config/planner.yaml" command="load"/> -->
        
            <rosparam file="$(find hide_and_seek)/config/costmap_common_$(arg second_tb3).yaml" command="load" ns="global_costmap" />
            <rosparam file="$(find hide_and_seek)/config/costmap_common_$(arg second_tb3).yaml" command="load" ns="local_costmap" />
        
            <rosparam file="$(find hide_and_seek)/config/costmap_local.yaml" command="load" ns="local_costmap" />
            <param name="local_costmap/width" value="20.0"/> <!-- 10.0 --> 
            <param name="local_costmap/height" value="20.0"/> <!-- 10.0 -->
        
            <!-- <rosparam file="$(find hide_and_seek)/config/costmap_global_static.yaml" command="load" ns="global_costmap" unless="$(arg no_static_map)"/> -->
            
            <rosparam file="$(find hide_and_seek)/config/costmap_global_laser.yaml" command="load" ns="global_costmap" if="$(arg no_static_map)"/>
            <param name="global_costmap/width" value="20.0" if="$(arg no_static_map)"/>
            <param name="global_costmap/height" value="20.0" if="$(arg no_static_map)"/>
        
            <remap from="/odom" to="$(arg second_tb3)/odom" />
            <remap from="/cmd_vel" to="$(arg second_tb3)/cmd_vel" />
        
          </node>
          <node pkg="hide_and_seek" type="seeker_bot.py" name="seeker_node" output="screen"/>
    </group>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find hide_and_seek)/rviz/config.rviz" />
</launch>